cssselect/parser.py

```diff
@@ -156,50 +165,45 @@ class FunctionalPseudoElement(object):
         Use at your own risks.
 
     """
-    def __init__(self, name, arguments):
+    def __init__(self, name: str, arguments: Sequence['Token']):
         self.name = ascii_lower(name)
         self.arguments = arguments
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return '%s[::%s(%r)]' % (
             self.__class__.__name__, self.name,
             [token.value for token in self.arguments])
 
-    def argument_types(self):
+    def argument_types(self) -> List[str]:
         return [token.type for token in self.arguments]
 
-    def canonical(self):
+    def canonical(self) -> str:
         args = ''.join(token.css() for token in self.arguments)
         return '%s(%s)' % (self.name, args)
 
-    def specificity(self):
```

Comment:
Is this intended to be deleted?

Reply:
Yes it is: it was never run: self.selector doesn't exist so it would crash if it was used. `FunctionalPseudoElement` is never used in a tree but separately so it doesn't need to have the same interface as the other classes around.